src/system-fetch.js

```diff
@@ -59,6 +59,29 @@
       }
     };
   }
+  else if (typeof fetch != 'undefined') {
```

Comment:
I'm just trying to think carefully about the environments here - if I had `fetch` defined as a global in Node, I probably wouldn't want it to be used either right? So perhaps we should drop it below the `typeof require != 'undefined'` check as well? Then the issue is if we're in a service worker with `require` defined, so perhaps we can add a `typeof process != 'undefined'` check as well to avoid that.